drivers: i2c: Add mcux flexcomm driver

MaureenHelm opened the pull request:
Adds a new shim driver for the flexcomm i2c, and enables it on lpcxpresso55s69 and lpcxpresso54114 boards.

~Depends on zephyrproject-rtos/hal_nxp#33~
~Depends on #23972 or #24012~

Fixes #22701

MaureenHelm committed:
dts: Add i2c bindings and nodes for lpc socs

Adds i2c device tree bindings and nodes for the lpc54xxx and lpc55s6x
socs in preparation for adding a new i2c driver.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
drivers: i2c: Add mcux flexcomm driver

Adds a shim layer around the mcux lpc flexcomm driver to adapt it to the
zephyr i2c interface. It leverages heavily from the existing mcux lpi2c
shim driver.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
soc: nxp_lpc: Enable the mcux flexcomm i2c driver

Enables the mcux flexcomm i2c driver on lpc54xxx and lpc55xxx socs.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
boards: lpcxpresso55s69: Enable i2c instance 4

Enables i2c instance 4 on the lpcxpresso55s69 board. Configures pinmuxes
and clocks, and updates board documentation accordingly.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
boards: lpcxpresso54114: Enable i2c instance 4

Enables i2c instance 4 on the lpcxpresso54114 board. Configures pinmuxes
and clocks, and updates board documentation accordingly.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

galak commented on drivers/i2c/i2c_mcux_flexcomm.c:
```diff
@@ -0,0 +1,254 @@
+		irq_enable(DT_INST_##id##_NXP_LPC_I2C_IRQ_0);		\
+	}								\
+
+#ifdef DT_INST_0_NXP_LPC_I2C
```
Make this DT_INST_FOREACH(I2C_MCUX_FLEXCOMM_DEVICE)

MaureenHelm committed:
samples: sensor: Add lpcxpresso55s69 to fxos8700 accel-only test

The lpcxpresso55s69 board has an mma8652 accelerometer which is
compatible with the fxos8700 driver in accel-only mode. Now that this
board has an i2c driver, include it in the platform whitelist.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>
